pages/dkp/kommander/1.4/projects/platform-services/platform-services-catalog/kudo-spark/index.md

```diff
@@ -0,0 +1,518 @@
+By default, it supports integration with the [Prometheus operator](https://github.com/coreos/prometheus-operator).
+
+Prometheus Operator relies on `ServiceMonitor` kind which describes the set of targets to be monitored.
+KUDO Spark Operator configures `ServiceMonitor`s for both the Operator and submitted Spark Applications automatically
+when monitoring is enabled.
+
+#### Exporting Spark Operator and Spark Application metrics to Prometheus
+
+##### Prerequisites
+* KUDO v0.15.0 or later
+* The `prometheus-operator`.
+  If you don't already have the `prometheus-operator` installed on your Kubernetes cluster, you can do so by following
+  the [quick start guide](https://github.com/coreos/prometheus-operator#quickstart).
+
+##### Metrics configuration
+Metrics reporting is disabled by default and can be enabled by passing the following parameter to `kudo install`:
+```bash
+$ kubectl kudo install spark --instance=spark -p enableMetrics=true
+```
+
+Service endpoints and ServiceMonitor resources are configured to work with Prometheus Operator
+out of the box.
+
+Full list of metrics configuration parameters and defaults is available in KUDO Spark [params.yaml](https://github.com/kudobuilder/operators/blob/master/repository/spark/operator/params.yaml).
+
+##### Running Spark Application with metrics enabled
+1) Composing your Spark Application yaml:
+    - use the following Spark image which includes the `JMXPrometheus` exporter jar: `mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s`
+    - enable Driver and Executors metrics reporting by adding the following configuration into `SparkApplication` `spec` section:
+      ```yaml
+        monitoring:
+          exposeDriverMetrics: true
+          exposeExecutorMetrics: true
+          prometheus:
+            jmxExporterJar: "/prometheus/jmx_prometheus_javaagent-0.11.0.jar"
+            port: 8090
+      ```  
+      `spec.momitoring.prometheus.port` value should be the same for all submitted Spark Applications in order for metrics to be scraped.
+    - if it's necessary to expose the metrics endpoint on a port other than `8090`, do the following:
+        - specify desired port when installing the `kudo-spark-operator`:
+
+      ```
+      kubectl kudo install spark -p appMetricsPort=<desired_port>
+      ```
+        - change the `port` value in the `SparkApplication` yaml definition (`spec.monitoring.prometheus.port`)
+    - Mark `driver` and/or `executor` with the label `metrics-exposed: "true"` -
+      ```yaml
+      spec:
+        driver:
+          labels:
+             metrics-exposed: "true"
+        executor:
+          labels:
+            metrics-exposed: "true"
+      ```
+    - Install the SparkApplication:
+      ```
+      kubectl apply -f <path_to_the_application_yaml>
+      ```
+   Full application configuration example is available in [spark-application-with-metrics.yaml](resources/monitoring/spark-application-with-metrics.yaml)
```

Comment:
@kaiwalyajoshi - this link is broken (same thing - weird on the kudo.dev site) - should this point here?

```suggestion
   Full application configuration example is available in [spark-application-with-metrics.yaml](https://github.com/kudobuilder/operators/blob/6f7fbecaf6a79e22328149062ea4d3e6dd4082b9/repository/spark/docs/2.4.5-1.0.1/resources/monitoring/spark-application-with-metrics.yaml)
```

Reply:
replaced with the correct link